mtags/src/main/scala-2/scala/meta/internal/pc/InferredMethodProvider.scala

```diff
@@ -0,0 +1,496 @@
+                      s"arg$index: ${prettyType(p)}"
+                    }
+                    .mkString(", ")
+                val resultType = args.last
+                val ret = prettyType(resultType)
+                signature(
+                  name = errorMethod.name.toString(),
+                  paramsString,
+                  Option(ret),
+                  identity,
+                  None
+                )
+
+              case _ =>
+                //    def method1(s : => Int) = 123
+                //    method1(<<otherMethod>>)
+                val ret = tpe.toString().replace("=>", "").trim()
+                val full =
+                  s"def ${errorMethod.name}: $ret = ???\n$indentString"
+                signature(
+                  name = errorMethod.name.toString(),
+                  "",
+                  None,
+                  _ => full,
+                  Option(lastApplyPos)
+                )
+            }
+
+          case _ =>
+            Nil
+        }
+      } else {
+        signature(
+          name = errorMethod.name.toString(),
+          paramsString = argumentsString(arguments).getOrElse(""),
+          retType = None,
+          postProcess = identity,
+          position = None
+        )
+      }
+    }
+    def makeEditsForListApplyWithoutArgs(
+        argumentList: Name,
+        errorMethod: Ident
+    ): List[(FileURI, TextEdit)] = {
+      val listSymbol = context.lookupSymbol(argumentList, _ => true)
+      if (listSymbol.isSuccess) {
+        listSymbol.symbol.tpe match {
+          // need to find the type of the value on which we are mapping
+          case TypeRef(_, _, TypeRef(_, inputType, _) :: Nil) =>
+            val paramsString = s"arg0: ${prettyType(inputType.tpe)}"
+            signature(
+              name = errorMethod.name.toString(),
+              paramsString,
+              None,
+              identity,
+              None
+            )
+          case NullaryMethodType(
+                TypeRef(_, _, TypeRef(_, inputType, _) :: Nil)
+              ) =>
+            val paramsString = s"arg0: ${prettyType(inputType.tpe)}"
+            signature(
+              name = errorMethod.name.toString(),
+              paramsString,
+              None,
+              identity,
+              None
+            )
+          case _ =>
+            Nil
+        }
+      } else Nil
+    }
+    def makeEditsMethodObject(
+        arguments: List[Tree],
+        container: Name,
+        errorMethod: Select
+    ): List[(FileURI, TextEdit)] = {
+      def makeClassMethodEdits(
+          template: Template
+      ): List[(FileURI, TextEdit)] = {
+        val insertPos: Position =
+          inferEditPosition(template)
+        val templateFileUri = template.pos.source.content
+          .map(_.toString)
+          .mkString
+        if (params.uri().toString() != getPositionUri(insertPos)) {
+          println("You can only infer method for current file.")
```

Comment:
good idea! Should I approach it like this for every spot where we return `Nil` as text edits?
My rationale is that in that case we will always fail weirdly, while it would be better to let the user know that is an unexpected case.
So I can create a utility to log/throw in all those places.